kedro/framework/project/__init__.py

```diff
@@ -259,3 +261,58 @@ def validate_settings():
         )
 
     importlib.import_module(f"{PACKAGE_NAME}.settings")
+
+
+def find_pipelines() -> Dict[str, Pipeline]:
+    """Automatically find modular pipelines having a ``create_pipeline``
+    function. By default, projects created using Kedro 0.18.3 and higher
+    call this function to autoregister pipelines upon creation/addition.
+
+    Returns:
+        A generated mapping from pipeline names to ``Pipeline`` objects.
+
+    Warns:
+        UserWarning: When a module does not expose a ``create_pipeline``
+            function, the ``create_pipeline`` function does not return a
+            ``Pipeline`` object, or if the module import fails up front.
+    """
+    pipelines_dict = {"__default__": pipeline([])}
+    for pipeline_dir in importlib_resources.files(
+        f"{PACKAGE_NAME}.pipelines"
+    ).iterdir():
+        if not pipeline_dir.is_dir():
+            continue
+        pipeline_name = pipeline_dir.name
+        if pipeline_name == "__pycache__":
+            continue
```

Comment:
This is fine, but just wondering whether there are there any other special cases we should just skip here? Or should we make it more general, e.g. skip anything that starts with `_` or `.`? Maybe there's even something build into pathlib or `importlib_resources` to skip any files that don't look promising?

Reply:
`__pycache__` is a weird case that gets generated automatically and is actually importable. I'm not sure what other scenarios would actually need to get skipped; technically, is there any reason we should stop somebody from defining a pipeline starting with underescore, for example?

Reply:
No, no particular reason. I didn't realise that `__pycache__` was actually importable so now understand the need to skip it!